### PR TITLE
BFieldSurfaceIntegral

### PR DESCRIPTION
**Pull request by saurabh-s-sawant**
Surface integral of a Bfield will be implemented here.

### CODE REVIEW

File: Source/Diagnostics/ReducedDiags/RawBFieldReduction.H [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                    amrex::Real y = 0._rt;
+                    amrex::Real fac_z = (1._rt - Bx_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - Bx_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - Bx_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    // we want to weight interior face-nodes by 1
```
Comment (ajnonaka) [inline Source/Diagnostics/ReducedDiags/RawBFieldReduction.H]: I would replace all instances of "face-nodes" with simply "faces".  Including the word 'node' in there could be misleading.  Something that is nodal in exactly one direction is "face" data; but in AMReX-speak we just refer to these as "faces".

Comment (saurabh-s-sawant) [reply Source/Diagnostics/ReducedDiags/RawBFieldReduction.H]: OK Andy I will make this change and push.

**Comment by saurabh-s-sawant** [comment]
[](url)

File: Docs/source/usage/parameters.rst [hunk 2]
```diff
@@ -2522,6 +2522,57 @@ Reduced Diagnostics
...
+        The output columns correspond to the timestep counter, physical time, and reduced values of Bx, By, Bz components.
+
+        * ``<reduced_diags_name>.integration_type`` (`string`)
+           The type of integration to be performed. It must be either ``surface`` or ``volume``.
+           Note that the surface integral provides the integrated normal as well as the
+           traverse fields over the surface. For the surface integral, the user-defined parser
+           for the plane should be less than one cell size in thickness in the direction of the surface normal.
+           Also, if the surface to be defined has an edge that overlaps with the domain boundary but the edge parallel
+           to it does not overlap with the domain boundary, e.g. a half cross-section of a plane, then exclude the
+           edge that overlaps with the domain boundary while defining the surface.
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst:2654]: is this valud for the B-field defined at the face-center. The normal component of B-field is on the face-center for yee

File: Docs/source/usage/parameters.rst [hunk 3]
```diff
@@ -2522,6 +2522,57 @@ Reduced Diagnostics
...
+           in the negative direction.
+
+        * ``<reduced_diags_name>.scaling_factor`` (`string`)  optional (default `1 1 1`)
+           This parameter is used when the ``integration_type`` is set to ``surface``. The parser takes three values to scale
+           the reduced field quantities, namely, the surface integral of Bx, By, and Bz.
+           This parameter can be used in the following two scenarios:
+           Let's say, we require the surface integral of Bx on a surface, with the surface normal in the negative x direction.
+           In this case, we would specify the value of this parameter as ``-1. 1. 1.`` so that the surface integral of Bx is multiplied by ``-1``.
+           As another example, we may require surface integral of H-field, which can be obtained, say, by dividing the surface integrals of Bx, By, and Bz
+           by a constant value of permeability, ``mu``. In this case, we would specify the value of this parameter as ``1./mu, 1./mu, 1./mu``.
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst]: are we assuming that `mu` will be constant?

File: Source/Diagnostics/ReducedDiags/RawBFieldReduction.H [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - Bx_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - Bx_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    // We want to weight interior faces by 1
+                    // and faces that are on the faces of the box by 0.5
+            // Faces for Bx computation can lie on xmin and xmax box faces.
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawBFieldReduction.H]:
```suggestion
                    // and faces that are on the faces of the box by 0.5
                    // Faces for Bx computation can lie on xmin and xmax box faces.
```

File: Source/Diagnostics/ReducedDiags/RawBFieldReduction.H [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - By_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - By_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    // We want to weight interior faces by 1
+                    // and faces that are on the faces of the box by 0.5
+            // Faces for By computation can lie on ymin and ymax box faces.
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawBFieldReduction.H]:
```suggestion
                    // and faces that are on the faces of the box by 0.5
                    // Faces for By computation can lie on ymin and ymax box faces.
```

File: Source/Diagnostics/ReducedDiags/RawBFieldReduction.H [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - Bz_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - Bz_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    // We want to weight interior faces by 1
+                    // and faces that are on the faces of the box by 0.5
+            // Faces for Bz computation can lie on zmin and zmax box faces.
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawBFieldReduction.H]:
```suggestion
                    // and faces that are on the faces of the box by 0.5
                    // Faces for Bz computation can lie on zmin and zmax box faces.
```

**Review by RevathiJambunathan — APPROVED** [review]
thanks for this PR, @PeculiarOvertones 
I have super minor comments

File: Docs/source/usage/parameters.rst (hunk 2 above)
Comment (saurabh-s-sawant) [reply Docs/source/usage/parameters.rst:2654]: Yes.

File: Docs/source/usage/parameters.rst (hunk 3 above)
Comment (saurabh-s-sawant) [reply Docs/source/usage/parameters.rst]: Yes

Comment (saurabh-s-sawant) [reply Docs/source/usage/parameters.rst]: The latest commit clarifies it

**Pull request merged**